src/openvic-simulation/economy/FactoryProducer.cpp

```diff
@@ -37,3 +44,64 @@ fixed_point_t FactoryProducer::get_average_profitability_last_seven_days() const
 
 	return sum / (1 + profit_history_current);
 }
+
+void FactoryProducer::produce(
+	const fixed_point_t input_modifier, const fixed_point_t throughput_modifier, const fixed_point_t output_modifier
+) {
+	std::map<PopType const* const, fixed_point_t> employees_per_job {};
```

Comment:
This could be cached as a private field.

Reply:
Yeah this definitely should be cached, we'll also need it for tooltips

Reply:
I think tooltips should be calculated on demand. It saves a memory and ensures the tooltips are up to date.